proto/warehouse/warehouse.proto

```diff
@@ -109,11 +113,69 @@ message RetryWHUploadsResponse {
 }
 
 message ValidateObjectStorageRequest {
-  string type=1;
-  google.protobuf.Struct config=2;
+  string type = 1;
+  google.protobuf.Struct config = 2;
 }
 
 message ValidateObjectStorageResponse{
-  bool isValid=2;
-  string error=3;
+  bool isValid = 2;
+  string error = 3;
+}
+
+message RetrieveFailedBatchesRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+}
+
+message RetrieveFailedBatchesResponse{
+  repeated FailedBatchInfo infos = 1;
+}
+
+message FailedBatchInfo{
+  string errorCategory = 1;
+  string sourceID = 2;
+  int64 count = 3;
+  google.protobuf.Timestamp lastHappened = 4;
+  string status = 5;
+}
+
+message RetrieveFailedBatchRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+  string errorCategory = 4;
+  string sourceID = 5;
+  string status = 6;
+}
+
+message RetrieveFailedBatchResponse{
+  string error = 1;
+  string errorCategory = 2;
+  string sourceID = 3;
+  google.protobuf.Timestamp lastHappened = 4;
+  string status = 5;
+}
+
+message RetryFailedBatchesRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+}
+
+message RetryFailedBatchesResponse{
+  int64 count = 1;
+}
+
+message RetryFailedBatchRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+  string errorCategory = 4;
+  string sourceID = 5;
+  string status = 6;
+}
+
+message RetryFailedBatchResponse{
+  int64 count = 1;
```

Comment:
Same question as above.

Reply:
It represents a number of retries syncs. Renaming it to `retriedSyncsCount`.